bugsnag/src/main/java/com/bugsnag/Bugsnag.java

```diff
@@ -81,7 +84,18 @@ private void scheduleSessionFlushes() {
         sessionExecutorService.scheduleAtFixedRate(new Runnable() {
             @Override
             public void run() {
-                sessionTracker.flushSessions(new Date());
+                // Create a new thread which is not a deamon thread to actually flush the sessions
+                // This will means that the thread will not be killed
+                // half way through if the application shuts down
+                Thread nonDeamonThread = new Thread(new Runnable() {
```

Comment:
We could use a cached executor e.g. `Executors.newSingleThreadExecutor()` rather than creating a new thread every time this is invoked.

Reply:
Added new newSingleThreadExecutor